routes/auth.routes.js

```diff
@@ -0,0 +1,32 @@
+const {Router} = require ('express')
+const router = new Router()
+const bcryptjs = require('bcryptjs')
+const saltRounds =10
+const User = require ('../models/User.model')
+
+// GET route ==> to display the signup form to users
+router.get ('/signup', (req,res) => res.render('views/signup.hbs'))
+
+// POST route ==> to process form data
+router.post('/signup', (req, res, next) => {
+console.log('The form data: ', req.body); 
+const {username, password} =req.body;
+  bcryptjs
+  .genSalt(saltRounds)
+  .then(salt => bcryptjs.hash(password, salt))
+  .then (hashedPassword => {
+  console.log(`Password hash: ${hashedPassword}`)
+  return User.create ({
+      username,
+      passwordHash: hashedPassword
+  })
+})
+.then(userFromDB =>{
+    console.log('Newly created user is: ', userFromDB);
+    res.redirect('/userProfile');  
+}) 
+.catch(error => next (error))
+})
+
+router.get ('/userProfile', (req, res) => res.render('/users/use-profile'))
```

Comment:
don't use a leading `/`; this should only be `users/user-profile` (again, careful with typos)